src/lib/keys.js

```diff
@@ -64,68 +75,100 @@ const genKey = (networkSeed, point, revision) => {
   return b64(jam(sed));
 };
 
-// 'next' refers to setting the next set of keys
-// if false, we use revision - 1
-const attemptNetworkSeedDerivation = async (next, args) => {
-  const { walletType, urbitWallet, authMnemonic } = args;
-
-  // NB (jtobin):
-  //
-  // following code is intentionally verbose for sake of clarity
-
-  const point = need.point(args.pointCursor);
-  const pointDetails = need.fromPointCache(args.pointCache, point);
-
-  const revision =
-    next === true
-      ? parseInt(pointDetails.keyRevisionNumber)
-      : parseInt(pointDetails.keyRevisionNumber) - 1;
-
-  let managementSeed = '';
-
-  const ticketLike = [WALLET_TYPES.TICKET, WALLET_TYPES.SHARDS];
-
-  if (ticketLike.includes(walletType)) {
-    const uwal = urbitWallet.matchWith({
-      Just: uw => uw.value,
-      Nothing: _ => {
-        throw new Error(BRIDGE_ERROR.MISSING_URBIT_WALLET);
-      },
-    });
+/**
+ * @param {object} urbitWallet
+ * @param {number} revision
+ * @return {Promise<Maybe<string>>}
+ */
+export const deriveNetworkSeedFromUrbitWallet = async (
+  urbitWallet,
+  revision = 1
+) => {
+  return await deriveNetworkSeedFromSeed(urbitWallet.management.seed, revision);
+};
 
-    managementSeed = uwal.management.seed;
-  } else if (walletType === WALLET_TYPES.MNEMONIC) {
-    const walProxy = need.addressFromWallet(args.wallet);
+/**
+ * @param {Maybe<any>} wallet
+ * @param {string} authMnemonic
+ * @param {object} details
+ * @param {number} revision
+ * @return {Promise<Maybe<string>>}
+ */
+export const deriveNetworkSeedFromMnemonic = async (
+  wallet,
+  authMnemonic,
+  details,
+  revision = 1
+) => {
+  const isOwner = eqAddr(wallet.address, details.owner);
+  const isManagementProxy = eqAddr(wallet.address, details.managementProxy);
+
+  // the network seed is only derivable from mnemonic if the derived
```

Comment:
is this comment (and impl) true? the original code only allowed it if the mnemonic was the management proxy but it seems to make sense that you should also be able to do this if you're the owner.